tests/integration/test_charm.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+from pathlib import Path
+
+import pytest
+import yaml
+
+log = logging.getLogger(__name__)
+
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+
+APP_NAME = "envoy"
+CHARM_ROOT = "."
+
+
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test):
+    charm = await ops_test.build_charm(".")
+    await ops_test.model.deploy(charm)
+    await ops_test.model.wait_for_idle(status="active")
```

Comment:
Check out the discussion [here](https://github.com/canonical/operator-template/issues/1), specifically the last code block.  I think adding `raise_on_blocked=True` is good for these, as hitting a `Blocked` (which should mean "I need human intervention") is a failure as it shouldn't have needed humans.  I think the `assert active` at the end might be superfluous now that `wait_for_idle` has `status='active'` arg, but not sure

Reply:
When adding the `raise_on_blocked=True` flag, this test fails. The envoy charm seems to have some difficulty accessing its oci-image on my machine, which blocks deployment for a few seconds before the image can be found, raising a transient BlockedStatus. This also slows down the second test, where it takes over 5 minutes to redeploy envoy based on its oci-image. Would you mind testing this locally, so I can verify if it's my setup that's causing issues? 

Also, the assertion is in the test_grpc_relation test. I left it in so we are assured that the environment is ready to define a relation before we do it, so when we later assert everything's active, we know that the changes are occurring as expected. It's an extra level of thoroughness, but I'm happy to remove it if you think it's unnecessary.

Reply:
I've added an issue [here](https://github.com/canonical/envoy-operator/issues/7) detailing the issues with raise_on_blocked, and added a link to the issue in a comment above this line.